src/sql/workbench/browser/designer/designerPropertyPathValidator.ts

```diff
@@ -0,0 +1,52 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { DesignerPropertyPath, DesignerTableProperties, DesignerViewModel } from 'sql/workbench/browser/designer/interfaces';
+
+export class DesignerPropertyPathValidator {
+
+	/**
+	 * Validate the path property, detail of the path can be found in the azdata typings file.
+	 * @param path path of the property.
+	 * @param viewModel the view model.
+	 * @returns Whether the path is valid.
+	 */
+	static validate(path: DesignerPropertyPath, viewModel: DesignerViewModel): boolean {
+		// the path must have items and currently we support up to 5 items in the path.
+		if (!path || path.length === 0 || path.length > 5) {
+			return false;
+		}
+
+		for (let index = 0; index < path.length; index++) {
+			const expectingNumber = (index % 2) !== 0;
```

Comment:
the logic here isn't self-explanatory to me.  maybe a comment would help why we expect number/strings on odd/even elements?

Reply:
added comment to explain all the supported scenarios.